checkstyle/import-control.xml

```diff
@@ -74,6 +74,7 @@
     <subpackage name="protocol">
       <allow pkg="org.apache.kafka.common.errors" />
       <allow pkg="org.apache.kafka.common.protocol.types" />
+      <allow pkg="org.apache.kafka.clients.consumer" />
```

Comment:
This seems like a red flag, doesn't it? There's an explicit disallow of `o.a.k.clients` in `o.a.k.common` earlier since common is supposed to be a lower layer than clients.

I haven't searched exhaustively, so maybe I missed others, but it looks like at least OffsetOutOfRangeException is a problem. We have to be careful about moving it since it is public API, but could we move the class, then replace the version in `o.a.k.clients` with a subclass of the one moved to `o.a.k.common`?